exonum/src/node/consensus.rs

```diff
@@ -553,7 +553,7 @@ impl NodeHandler {
             .merge(fork.into_patch())
             .expect("Unable to save transaction to persistent pool.");
 
-        if self.state.is_leader() {
+        if self.state.is_leader() && self.state.round() != Round::zero() {
```

Comment:
The problem I encountered:
1. Node starts with 40_000 transactions in pool for verification.
2. Because of this, `maybe_add_propose_timeout` is triggered.
3. Since round = 0, panic occurs when trying to get `round.previous()`.

I decided that `self.maybe_add_propose_timeout()` deserved an additional check.

Reply:
Mmm... Digging through the callee tree of `NodeHandler::maybe_add_propose_timeout()` I see that `Round::previous()` is called only from `NodeHandler::round_start_time()`. I guess it may be better to teach `round_start_time()` to handle this situation correctly (just return `height_start_time()` I think) rather than tacking this ad-hoc fix on here.

Reply:
That's a very good point and needs to be discussed. I agree that fix proposed by me is very ad-hoc'y. But I think the problem is much deeper: reckless attitude towards error handling in Exonum. We have a ton of `assert`s and `panic!`s all around the non-test code which could be replaced by `Option` or `Result` (`Round::previous` is a great example of this). If I were to move this fix somewhere else, I would do so by making `Round::previous` return `Option` rather than teaching `round_start_time()` to do things it is not responsible for.

Reply:
@vldm I know you are not a fan of `panic`s, what do you think of this situation?

Reply:
Only genesis height starts from zero round, other started from first.

Maybe `State` should initialize with first round?
@alekseysidorov @slowli

Reply:
@andreyluchin I'd not classify all `assert`s / `panic`s as necessarily reckless; in fact, I think the current codebase may *lack* them in consensus and other safety-critical code. If used properly, `assert`s verify invariants or (if `debug_assert` is used) to signal them to the developers. OTOH, we shouldn't `panic` when an error can be expected.

Re: 0th round. If I understand correctly, `Round::zero()` is used in code mostly as an ad-hoc replacement for `None`, allowing to declare some fields (e.g., the locked round of the node, or the latest known round of peers) as `Round` instead of `Option<Round>`. I'm not sure why this is the case, but I'd argue that this "optimization" hurts code maintainability.

Reply:
I think that returning height_start_time for Round(0) may cause confusion. 
Maybe round should be replaced to store inner value as NonZeroU32 since 0 is logically invalid value for round number.
For lock round Option<Round> may be used to indicate that there can be no lock.

Reply:
https://exonum.com/doc/advanced/consensus/specification/#rounds
>Rounds are numbered from 1

This does indeed seem like a case for using placeholder values instead of `Option`. I believe we had (or still) have similar thing for IP address where `0.0.0.0` indicates its absence.

Anyway, fixing this is out of the scope of this PR for sure. But I would definitely create an issue for reworking `Round` if I get @DarkEld3r approval.